chapter/related-work.md

```diff
@@ -6,6 +6,90 @@ As such, it is important to get a picture of the field of current LSP projects.
+
+The Monto IR is a language agnostic and editor independent, JSON serialized tree-like model.
+Additionally, the IR maintains low level syntax highlighting information (font, color, style, etc.) but leaves the highlighting to the language specific service.
+
+The processing and modification of the source code and IR is performed by *Monto Services*.
+Services implement specific actions, e.g. parsing, outlining or highlighting.
+A central broker connects the services with each other and the editor.
+
+Since Monto performs all work on the IR, independent of the editor, and serializes the IR as JSON messages, the language used to implement *Monto Services* can be chosen freely giving even more flexibility.
+
+The Editor extension's responsibility is to act as a source and sink for data.
+It sends Monto compliant messages to the broker and receives processing results such as (error) reports.
+The communication is based on the ZeroMQ[zeromq] technology which was chosen because it is lightweight and available in manly languages [@monto-disintegrated] allowing to make use of existing language tools.
+
+#### Merlin
+
+The Merlin tool [@merlin,@merlin-website] is in many ways a more specific version of the idea presented in Monto.
+Merlin is a language server for the Ocaml language, yet predates the Language Server Protocol.
+
+The authors of Merlin postulate that implementing "tooling support traditionally provided by IDEs" for "niche languages" demands to "share the language-awareness logic" between implementations.
+As an answer to that, they describe the architecture of Merlin in [@merlin].
+
+Similarly to Monto, Merlin separates editor extensions from language analysis.
+Conversely, its interaction builds on a command line interface instead of message passing.
+Editor extensions expose the server functions to the user by integrating with the editor.
+
+The Merlin server hand provides a single optimized implementation of code intelligence for Ocaml.
+Since all resources could be put to a single project, multiple iterations of performance improvements were done on Merlin.
+It now supports partial, incremental parsing and type-checking which allows users to query information even about incomplete or incorrect programs.
+
+Notably, being written in Ocaml, Merlin can make use of existing tools of the Ocaml language.
+In fact, its parser and type-checker are based on the respective original implementations.
+The Merlin project did however have to adapt the Ocaml type-checker to support the aforementioned incrementality.
+Changes are made against a copy of the relevant modules shipped with Merlin which facilitates keeping up with the latest developments of the language.
+
+While Merlin serves as a single implementation used by all clients, unlike Monto it does not specify a language independent format, or service architecture.
```

Comment:
Maybe repeat here also the big difference that it specializes in only one language, while Monto is generic, which is a huge difference.

Reply:
```suggestion
While Merlin serves as a single implementation used by all clients, unlike Monto it does not specify a language independent format, or service architecture.
In fact, Merlin explicitly specializes in a single language and provides a complete implementation where Monto merely defines the language agnostic interface to implement a server on.
```

Reply:
how about so or do you think it should be even more explicit?

Reply:
This sentence is fine.